examples/policies/l3/host/host-policy.json

```diff
@@ -0,0 +1,41 @@
+[
+  {
```

Comment:
can you add an example of a k8s CNP in yaml as well?

Reply:
It looks like I do need a couple changes to enable `nodeSelector` in CNPs :-/
```
$ cat examples/policies/l3/host/host-policy.yaml
apiVersion: "cilium.io/v2"
kind: CiliumNetworkPolicy
metadata:
  name: "allow-dns-on-udp-53-only"
spec:
  nodeSelector:
    matchLabels:
      {}
  ingress:
  - toPorts:
    - ports:
      - port: "6443"
        protocol: TCP
      - port: "22"
        protocol: TCP
  egress:
  - toPorts:
    - ports:
      - port: "4240"
        protocol: TCP
      - port: "8080"
        protocol: TCP
  - toCIDR:
    - "8.8.0.0/16"
    toPorts:
    - ports:
      - port: "53"
        protocol: UDP
$ kubectl create -f examples/policies/l3/host/host-policy.yaml 
The CiliumNetworkPolicy "allow-dns-on-udp-53-only" is invalid: spec.endpointSelector: Required value
```
I haven't looked at how much changes are required, but maybe best to address in a follow up?

Reply:
I don't mind getting the datapath bits in here first but obviously being able to install these policies with CNP is important before v1.8 full release. I don't expect this to be a big change.

Reply:
I took a second look at this and found a couple things to fix, but I'm still having that same error message. I think I'll need to move this to follow ups :-/